app/cost_service.py

```
@@ -0,0 +1,874 @@
+                    if exchange_rate_for_date:
+                        exchange_rate_month = exchange_rate_for_date['year_month']
+                        exchange_rate_value = Decimal(str(exchange_rate_for_date['rate']))
+                        logger.info(f"Found exchange rate for date {due_date}: {exchange_rate_value}")
+                        
+                        if converted_amount_brl is None:
+                            converted_amount_brl = amount * exchange_rate_value
+                    else:
+                        logger.warning(f"No exchange rate found for {currency} to BRL for period {exchange_rate_month}")
+                        
+                        
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)